src/player.h

```diff
@@ -110,11 +111,19 @@ class Player
 
 	DISABLE_CLASS_COPY(Player);
 
-	virtual void move(f32 dtime, Environment *env, f32 pos_max_d)
-	{}
-	virtual void move(f32 dtime, Environment *env, f32 pos_max_d,
-			std::vector<CollisionInfo> *collision_info)
-	{}
+	void step(float dtime, const PlayerControl &control,
+			const PlayerSettings &player_settings, Environment *env,
+			std::vector<CollisionInfo> *player_collisions = NULL);
```

Comment:
The code doesn't seem to account for `player_collisions == NULL` at all, why does it default to this?

also: should be `nullptr`

Reply:
Not sure, I'll need to check.

Reply:
Got it: the `Player::step` function does not care about the `CollisionInfo`, but this is used as a parameter when calling `Player::move`. This function does account for a null pointer.

Please keep in mind that all this PR is trying to do is lift the existing code and algorithms from `LocalPlayer` to `Player`. So answering for, or even changing, the existing movement code and call chain is outside of the scope of this PR.